docs/source/compression/compression_evaluator.rst

```diff
@@ -0,0 +1,305 @@
+Compression Evaluator
+=====================
+
+The ``Evaluator`` is used to package the training and evaluation process for models with similar topologies.
+To explain why NNI needs an ``Evaluator``, let's first look at the general process of model compression in NNI.
+
+In model pruning, some algorithms need to prune according to some intermediate variables (gradients, activations, etc.) generated during the training process,
+and some algorithms need to gradually increase or adjust the sparsity of different layers during the training process,
+or adjust the pruning strategy according to the performance changes of the model during the pruning process.
+
+In model quantization, NNI has quantization-aware training algorithm,
+it can adjust the scale and zero point required for model quantization from time to time during the training process,
+and may achieve a better performance compare to post-training quantization.
+
+In order to better support the above algorithms' need and maintain the consistency of the interface,
+NNI introduces the ``Evaluator`` as the carrier of the training and evaluation process.
+
+.. note::
+    For users prior to NNI v2.8: NNI previously provided APIs like ``trainer``, ``traced_optimizer``, ``criterion``, ``finetuner``.
+    These APIs were maybe tedious in terms of user experience. Users need to exchange the corresponding API frequently if they want to switch compression algorithms.
+    ``Evaluator`` is an alternative to the above interface, users only need to consider one parameter to support NNI to complete the compression process.
+
+For users of native Pytorch, ``TorchEvaluator`` requires the user to encapsulate the training process as a function and specifies the exposed interface,
+which will bring some complexity. But don't worry, in most cases, this will not change too much code.
+
+For users of `PytorchLightning <https://www.pytorchlightning.ai/>`__, ``LightningEvaluator`` can be created with only a few lines of code, and there is no need to make changes to the original code in most cases.
+
+Here we give two examples of how to create an ``Evaluator`` for both native Pytorch and PytorchLightning users.
+
+TorchEvaluator
+--------------
+
+``TorchEvaluator`` is for the users who work in a native Pytorch environment. Here is an example for how to initialize a ``TorchEvaluator``.
```

Comment:
Still didn't explain what is expected from the `training_func` here.

Reply:
add
`Please make sure each input arguments of the training_func is actually used,
especially max_steps and max_epochs can correctly control the duration of training.`
after this line, do you mean this?

Reply:
add more explanation before example, please review the latest version